Dockerfile.rhel7

```diff
@@ -0,0 +1,18 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
```

Comment:
Do we need a separate `Dockerfile.rhel7`?

Reply:
Maybe not.  The only open question is can we build the "upstream" version with only one docerfile.  though by the looks of it they are essentually the same.  Thoughts?  I have no reservations at this time to remove it.

Reply:
There is no "upstream" vs. "downstream" now.  There are only images built in api.ci, and developers can build images using `Dockerfile` based on UBI.